test/gpu_test.cpp

```diff
@@ -160,3 +164,105 @@ TEST(gpu_test, memory_manager_seperate_workspaces_allocsize)
     emitter.allocate_primitive_memory();
     EXPECT_EQ(emitter.sizeof_device_allocation(), total_size);
 }
+
+TEST(gpu_test, topk_3d_global_mem)
+{
+    Shape shape{4, 3073, 5};
+    auto A_cpu = make_shared<op::Parameter>(element::f32, shape);
+    auto A_gpu = make_shared<op::Parameter>(element::f32, shape);
+
+    auto B_cpu = make_shared<op::TopK>(A_cpu, 1, element::i64, 10, true);
+    auto B_gpu = make_shared<op::TopK>(A_gpu, 1, element::i64, 10, true);
+
+    auto cpu_f_0 = make_shared<Function>(make_shared<op::GetOutputElement>(B_cpu, 0),
+                                         op::ParameterVector{A_cpu});
+    auto cpu_f_1 = make_shared<Function>(make_shared<op::GetOutputElement>(B_cpu, 1),
+                                         op::ParameterVector{A_cpu});
+
+    auto gpu_f_0 = make_shared<Function>(make_shared<op::GetOutputElement>(B_gpu, 0),
+                                         op::ParameterVector{A_gpu});
+    auto gpu_f_1 = make_shared<Function>(make_shared<op::GetOutputElement>(B_gpu, 1),
+                                         op::ParameterVector{A_gpu});
+
+    test::Uniform<float> rng(0.0f, 1.0f);
+    vector<vector<float>> args_index;
+
+    for (shared_ptr<op::Parameter> param : cpu_f_0->get_parameters())
+    {
+        vector<float> tensor_val(shape_size(param->get_shape()));
```

Comment:
You really don't want to use random values because you end up with tests that sometimes fail, and then you can never get them to fail when you run them by hand.